app/azure/Dashboard/DashboardView.tsx

```diff
@@ -0,0 +1,70 @@
+import { useEnvironmentId } from '@/portainer/hooks/useEnvironmentId';
+import { PageHeader } from '@/portainer/components/PageHeader';
+import { DashboardItem } from '@/portainer/components/DashboardItem';
+import { error as notifyError } from '@/portainer/services/notifications';
+import PortainerError from '@/portainer/error';
+import { r2a } from '@/react-tools/react2angular';
+
+import { useResourceGroups, useSubscriptions } from '../queries';
+
+export function DashboardView() {
+  const environmentId = useEnvironmentId();
+
+  const subscriptionsQuery = useSubscriptions(environmentId);
+  if (subscriptionsQuery.isError) {
+    notifyError(
+      'Failure',
+      subscriptionsQuery.error as PortainerError,
+      'Unable to retrieve subscriptions'
+    );
+  }
+
+  const resourceGroupsQuery = useResourceGroups(
+    environmentId,
+    subscriptionsQuery.data
+  );
+  if (resourceGroupsQuery.isError && resourceGroupsQuery.queryErrors) {
+    resourceGroupsQuery.queryErrors.forEach((e) =>
+      notifyError(
+        'Failure',
+        e.error as PortainerError,
+        `Unable to retrieve resource groups for ${e.subscriptionId} resource group`
+      )
+    );
+  }
+
+  const isLoading =
+    subscriptionsQuery.isLoading || resourceGroupsQuery.isLoading;
+  if (isLoading) {
+    return null;
+  }
+
+  return (
+    <>
+      <PageHeader title="Home" breadcrumbs={[{ label: 'Dashboard' }]} />
+
+      {subscriptionsQuery.data && (
```

Comment:
data for both should be available here. I'd say move these to the lines before the return. so:
```
const subscriptionsCount = subscriptionsQuery.data.length
const resourceGroupCount = Object.values(resourceGroupsQuery.resourceGroups).flatMap((x) =>
                  Object.values(x)
                ).length
```

and then:
```
 <DashboardItem
            value={subscriptionCount}
            icon="fa fa-th-list"
            comment="Subscriptions"
          />
          <DashboardItem
              value={
                resourceGroupCount
              }
              icon="fa fa-th-list"
              comment="Resource groups"
            />
          ```

Reply:
done, I had to use ? to ensure the objects existed and define the counts `as number` as typescript was giving errors